packages/jest-circus/src/utils.ts

```diff
@@ -177,9 +179,17 @@ export const callAsyncCircusFn = (
     // soon as `done` called.
     if (fn.length) {
       let returnedValue: unknown = undefined;
+
       const done = (reason?: Error | string): void => {
+        if (testOrHook.seenDone) {
+          throw new Error('Expected done to be called once, but it was called multiple times.');
+        } else {
+          testOrHook.seenDone = true;
+        }
+
         // We need to keep a stack here before the promise tick
         const errorAtDone = new Error();
```

Comment:
```suggestion
        const errorAtDone = new ErrorWithStack(undefined, done);
```

a bit cleaner stack trace probably 👍 

(this one can be included in the change discussed here: https://github.com/facebook/jest/pull/10624/files#r503541181)